### PR TITLE
Add enum value headers

### LINKED ISSUE #1569: [rendering] Enum values not sufficiently separate from attribute footnotes
### Area(s)

area:other

## Description

When markdown is rendered, the attributes table is followed by footnotes, then by tables which list values for enums. Those enum tables are not sufficiently separated from the footnotes, and they _appear_ to be a part of the footnotes when reading. For example in [HTTP client semconv](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md#http-client) the `error.type` table of possible values _appears_ to be part of footnote **[14]** even though it is not.

## Example

Below is an example taken from HTTP client semconv that shows what I'm talking about

<details>
<summary>View example markdown</summary>

```markdown
**[14]:** The `url.template` MUST have low cardinality. It is not usually available on HTTP clients, but may be known by the application or specialized HTTP instrumentation.

The following attributes can be important for making sampling decisions
and SHOULD be provided **at span creation time** (if provided at all):

* [`http.request.method`](/docs/attributes-registry/http.md)
* [`server.address`](/docs/attributes-registry/server.md)
* [`server.port`](/docs/attributes-registry/server.md)
* [`url.full`](/docs/attributes-registry/url.md)

`error.type` has the following list of well-known values. If one of them applies, then the respective value MUST be used; otherwise, a custom value MAY be used.

| Value  | Description | Stability |
|---|---|---|
| `_OTHER` | A fallback error value to be used when the instrumentation doesn't define a custom value. | ![Stable](https://img.shields.io/badge/-stable-lightgreen) |
```
</details>

<details>
<summary>Above example rendered</summary>

> **[14]:** The `url.template` MUST have low cardinality. It is not usually available on HTTP clients, but may be known by the application or specialized HTTP instrumentation.
> The following attributes can be important for making sampling decisions
> and SHOULD be provided **at span creation time** (if provided at all):
> * [`http.request.method`](/docs/attributes-registry/http.md)
> * [`server.address`](/docs/attributes-registry/server.md)
> * [`server.port`](/docs/attributes-registry/server.md)
> * [`url.full`](/docs/attributes-registry/url.md)
> 
> `error.type` has the following list of well-known values. If one of them applies, then the respective value MUST be used; otherwise, a custom value MAY be used.
> | Value  | Description | Stability |
> |---|---|---|
> | `_OTHER` | A fallback error value to be used when the instrumentation doesn't define a custom value. | Stable |
</details>

What I'm asking for is some visual separation between these elements to make it more obvious that you have moved onto a separate section like this:

<details>
<summary>View example of suggested improved rendering (heading)</summary>

> **[14]:** The `url.template` MUST have low cardinality. It is not usually available on HTTP clients, but may be known by the application or specialized HTTP instrumentation.
> The following attributes can be important for making sampling decisions
> and SHOULD be provided **at span creation time** (if provided at all):
> * [`http.request.method`](/docs/attributes-registry/http.md)
> * [`server.address`](/docs/attributes-registry/server.md)
> * [`server.port`](/docs/attributes-registry/server.md)
> * [`url.full`](/docs/attributes-registry/url.md)
> ### Enum Values
> `error.type` has the following list of well-known values. If one of them applies, then the respective value MUST be used; otherwise, a custom value MAY be used.
> | Value  | Description | Stability |
> |---|---|---|
> | `_OTHER` | A fallback error value to be used when the instrumentation doesn't define a custom value. | Stable |
</details>

<details>
<summary>View example of suggested improved rendering (horizontal line separator)</summary>

> **[14]:** The `url.template` MUST have low cardinality. It is not usually available on HTTP clients, but may be known by the application or specialized HTTP instrumentation.
> 
> The following attributes can be important for making sampling decisions
> and SHOULD be provided **at span creation time** (if provided at all):
>
> * [`http.request.method`](/docs/attributes-registry/http.md)
> * [`server.address`](/docs/attributes-registry/server.md)
> * [`server.port`](/docs/attributes-registry/server.md)
> * [`url.full`](/docs/attributes-registry/url.md)
> 
> ---
> 
> `error.type` has the following list of well-known values. If one of them applies, then the respective value MUST be used; otherwise, a custom value MAY be used.
> 
> | Value  | Description | Stability |
< |---|---|---|
> | `_OTHER` | A fallback error value to be used when the instrumentation doesn't define a custom value. | Stable |
</details>

### Semantic convention version

v1.28.0

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dyladan**
Add headers in order to separate enum value tables from the footnotes above.

Fixes #1569

### CODE REVIEW

**Comment by jsuereth** [comment]
FYI - this one has poor interaction with TOC generation, which is a separate tool from table generation.  You could update the makefile to run toc generation *immediately after* weaver table generation.

**Comment by dyladan** [comment]
@jsuereth seems like `make fix` has correct behavior? The PR was wrong because I was using `make table-generation` directly instead of `make fix`

File: docs/http/http-spans.md [hunk 1]
```diff
@@ -17,6 +17,9 @@ and various HTTP versions like 1.1, 2 and SPDY.
 - [Name](#name)
 - [Status](#status)
 - [HTTP client](#http-client)
+  - [`error.type` Values](#errortype-values)
```
Comment (dyladan) [inline docs/http/http-spans.md:20]: @jsuereth I think this looks right

**Comment by dyladan** [comment]
Maybe we can add something fancier later but I'd like to get separators of some kind in and the other PR seems easier at the moment

**Pull request closed without merging**